### PR TITLE
Fixes a bug in `TensorStridedSliceUpdate`

### PR DESCRIPTION
**Pull request by eaplatanios**
@alextp Fixes a bug introduced in #27327 .

### CODE REVIEW

File: tensorflow/core/kernels/strided_slice_op.cc [hunk 1]
```diff
@@ -529,7 +529,6 @@ REGISTER_KERNEL_BUILDER(Name("ResourceStridedSliceAssign")
 REGISTER_KERNEL_BUILDER(Name("TensorStridedSliceUpdate")
                             .Device(DEVICE_GPU)
                             .TypeConstraint<int32>("T")
-                            .HostMemory("input")
```
Comment (alextp) [inline tensorflow/core/kernels/strided_slice_op.cc]: I think the int32 gpu kernel should have hostmemory annotations to be compatible with the other int32 gpu kernels

Comment (eaplatanios) [reply tensorflow/core/kernels/strided_slice_op.cc]: You're right. I reverted that change.

**Review by alextp — DISMISSED** [review]

**Comment by eaplatanios** [comment]
@alextp Sorry I had to make an additional small change to pass the backwards compatibility test. This is necessary because I forgot to add the output definition in the op registration before.

**Review by alextp — APPROVED** [review]

**Comment by alextp** [comment]
You shouldn't need to make the change to op_history.pbtxt because adding an output is backward-compatible and manual edits to that file are only required for non-backwards-compatible changes.

That said it's not harmful.

**Comment by eaplatanios** [comment]
> You shouldn't need to make the change to op_history.pbtxt because adding an output is backward-compatible and manual edits to that file are only required for non-backwards-compatible changes.

Thanks Alex! It's good to know that. The reason I actually did it is because this edit is technically not backwards-compatible as it changes the signature of the `TensorStridedSliceAssign` op (I had previously forgotten to declare that it has an output), thus resulting in a test failure.

**Comment by alextp** [comment]
Adding an output to an op is a backwards-compatible change though as it does not change the behavior of existing graphdefs.

**Comment by eaplatanios** [comment]
I see. Shouldn't the backwards compatibility check test take that into account then? Because it was failing until I made that change to `op_history.pbtxt` which added the output declaration.

**Pull request merged**